src/pages/settings/Profile/PersonalDetails/AddressPage.js

```diff
@@ -52,179 +52,159 @@ const defaultProps = {
     },
 };
 
-class AddressPage extends Component {
-    constructor(props) {
-        super(props);
-
-        this.validate = this.validate.bind(this);
-        this.updateAddress = this.updateAddress.bind(this);
-        this.onCountryUpdate = this.onCountryUpdate.bind(this);
-
-        const currentCountry = lodashGet(this.props.privatePersonalDetails, 'address.country') || '';
-        const currentCountryISO = PersonalDetails.getCountryISO(currentCountry) || CONST.COUNTRY.US;
-        const zipSampleFormat = lodashGet(CONST.COUNTRY_ZIP_REGEX_DATA, [currentCountryISO, 'samples'], '');
-        this.state = {
-            isUsaForm: currentCountry === CONST.COUNTRY.US || currentCountry === CONST.USA_COUNTRY_NAME,
-            zipFormat: this.props.translate('common.zipCodeExampleFormat', {zipSampleFormat}),
-        };
-    }
+/**
+ * @param {Function} translate - translate function
+ * @param {Boolean} isUsaForm - selected country ISO code is US
+ * @param {Object} values - form input values
+ * @returns {Object} - An object containing the errors for each inputID
+ */
+const validate = (translate, isUsaForm, values) => {
```

Comment:
Nothing wrong with this approach, but I'm just noting that this is a bit different than what we've been doing. We usually keep these functions inside the component and wrap them with useMemo/useCallback hooks.

Reply:
Ok, that sounds good to me, but what about functions that don't depend on props? I think those should be kept outside.